### PR TITLE
Pasting code into empty file  will trigger organize imports

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Issue #1075

Signed-off-by: Nikolas Komonen <nikolaskomonen@gmail.com>

### CODE REVIEW

**Comment by fbricon** [comment]
There should be a preference to enable or disable this feature

**Comment by fbricon** [comment]
Can you investigate whether it'll be possible to also automatically apply the quickfixes for these 2 errors that'll likely pop up:
- The declared package "foo.bar" does not match the expected package "org.acme"Java(536871240) -> change package declaration to org.acme
- The public type Foo must be defined in its own file Java(16777541)-> Rename type to Something (i.e if the code was copied into a file named Something.java)

**Comment by fbricon** [comment]
when adding a preference, don't forget to update the readme

**Comment by NikolasKomonen** [comment]
@fbricon I updated it with the preference java.paste.organizeImportsOnPaste

File: README.md [hunk 1]
```diff
@@ -116,6 +116,7 @@ The following settings are supported:
 * `java.codeGeneration.toString.listArrayContents`: List contents of arrays instead of using native toString(). Defaults to `true`.
 * `java.codeGeneration.toString.limitElements`: Limit number of items in arrays/collections/maps to list, if 0 then list all. Defaults to `0`.
 * `java.selectionRange.enabled`: Enable/disable Smart Selection support for Java. Disabling this option will not affect the VS Code built-in word-based and bracket-based smart selection.
+* `java.paste.organizeImportsOnPaste`: Organize imports is triggered when java code is pasted into an empty file.
```
Comment (fbricon) [inline README.md]: "java.actionsOnPaste.organizeImports": triggers "Organize imports" when java code is pasted into an empty file.

File: package.json [hunk 2]
```diff
@@ -388,6 +388,12 @@
           "default": true,
           "description": "Enable/disable Smart Selection support for Java. Disabling this option will not affect the VS Code built-in word-based and bracket-based smart selection.",
           "scope": "window"
+        },
+        "java.paste.organizeImportsOnPaste": {
```
Comment (fbricon) [inline package.json]: "java.actionsOnPaste.organizeImports"

File: package.json [hunk 3]
```diff
@@ -388,6 +388,12 @@
...
+          "type": "boolean",
+          "description": "Run organize imports when java code is pasted into an empty document.",
```
Comment (fbricon) [inline package.json]: triggers "Organize imports" when java code is pasted into an empty file.

File: src/settings.ts [hunk 4]
```diff
@@ -14,10 +14,13 @@ const changeItem = {
 };
 
 const EXCLUDE_FILE_CONFIG = 'configuration.checkProjectSettingsExclusions';
+export const ORGANIZE_IMPORTS_ON_PASTE = 'paste.organizeImportsOnPaste'; // java.paste.organizeImports
```
Comment (fbricon) [inline src/settings.ts]: "actionsOnPaste.organizeImports"

**Comment by NikolasKomonen** [comment]
@fbricon updated with new preference name

**Comment by fbricon** [comment]
Thanks @NikolasKomonen !

**Pull request merged**